tlatools/org.lamport.tlatools/test/tlc2/util/BufferedRandomAccessFileTest.java

```diff
@@ -170,4 +168,155 @@ public void testInvalidateBufferedData() throws IOException {
 			Assert.assertEquals(30, raf.read());
 		}
 	}
+
+	/**
+	 * Create an empty file, call seek(1), then try to read.  The correct behavior is to return -1 for EOF.
+	 * Earlier implementations of {@link BufferedRandomAccessFile} would incorrectly throw exceptions or return 0.
+	 */
+	@Test
+	public void testReadAfterSeekPastEndOfFile() throws IOException {
+		final File tmpFile = File.createTempFile("BufferedRandomAccessFileTest_testReadAfterSeekPastEndOfFile", ".bin");
+		tmpFile.deleteOnExit();
+		try (final java.io.RandomAccessFile raf = new BufferedRandomAccessFile(tmpFile, "r")) {
+			raf.seek(1);
+
+			assertEquals(1, raf.getFilePointer());
+
+			int charRead = raf.read();
+			assertEquals(-1, charRead);
+
+			byte[] buffer = new byte[100];
+			int nread = raf.read(buffer, 0, buffer.length);
+			assertEquals(-1, nread);
+		}
+	}
+
+	/**
+	 * Create an empty file, call seek(10000), then try to write.  The correct behavior is to extend the length of
+	 * the file and write the bytes.  Earlier implementations of {@link BufferedRandomAccessFile} would incorrectly
+	 * throw exceptions.
+	 */
+	@Test
+	public void testWriteAfterSeekPastEndOfFile() throws IOException {
+		final File tmpFile = File.createTempFile("BufferedRandomAccessFileTest_testWriteAfterSeekPastEndOfFile", ".bin");
+		tmpFile.deleteOnExit();
+		try (final java.io.RandomAccessFile raf = new BufferedRandomAccessFile(tmpFile, "rw")) {
+			raf.seek(10000);
+
+			assertEquals(10000, raf.getFilePointer());
+
+			byte[] buffer = new byte[100];
+			raf.write(buffer);
+			assertEquals(10000 + buffer.length, raf.length());
+			assertEquals(10000 + buffer.length, raf.getFilePointer());
+		}
+	}
+
+	@Test
+	public void regressionTest01() throws IOException {
```

Comment:
nit: I personally like giving these clearer names.  It's easier to read them if you have that context of what they intend to validate (or what you broke if they fail!).

Reply:
There's a lot to say about these `regressionTest*` methods that should be written in a comment.

TL;DR: These were generated by a computer, so they don't look particularly sensible, nor can I really assign them sensible names. I don't plan to change them (see below), but I am open to adding additional focused handwritten tests inspired by them.

----

The `regressionTest*` methods are all randomly-generated failures discovered by the fuzzer at some point while I was working on this PR. I copied them unchanged from the fuzzer output (although in a few cases I added a comment about what should happen for a particular test case).

The fact that these cases failed during development indicates they are testing _something_ that is tricky to get right in the implementation, and so I think it's worth having them here to make sure they are tested regularly.

I don't particularly want to change them because:
 - I am not confident I can minimize them reliably; the failures happened on intermediate versions of `BufferedRandomAccessFile` that are now lost. If I make a change and the test still passes, is it _really_ testing the same thing it was before?

And I don't want to remove them either:
 - Some of these came from an earlier version of the fuzzer (i.e. the current fuzzer doesn't cover them exactly as written here).
 - Small changes to the fuzzer generally result in large changes to explored traces (i.e. future versions of the fuzzer may not explore these specific cases).

Reply:
Sounds good, like a i said, no deal breakers for me at this point :)  I think the test updates you made are already above and beyond!